client/src/components/SharedComponents/Signup/index.js

```diff
@@ -0,0 +1,191 @@
+import React, { Component } from 'react'
+import './signup.css'
+import axios from 'axios'
+
+class Signup extends Component {
+  state = {
+    name: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+    displayBio: false,
+    type: '',
+    availability: '',
+    photo: '',
+    age: '',
+    description: '',
+    phone: '',
+    userType: 'user'
+  }
+  onChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  ShowDisplayBio = () => {
+    this.setState({ displayBio: true, userType: 'guide' })
+  }
+
+  displayShorterBio = () => {
+    this.setState({ displayBio: false })
+  }
+
+  validateForm = () => {
+    return (
+      this.state.name.length > 1 &&
```

Comment:
You can do this using a package, would be easier! There's a package called [yup](https://medium.com/@rossbulat/introduction-to-yup-object-validation-in-react-9863af93dc0e) you can use